tsne/bh_sne_src/sptree.cpp

```diff
@@ -259,26 +247,32 @@ template<int NDims>
 void SPTree<NDims>::subdivide() {
 
     // Create new children
-    double new_corner[NDims];
-    double new_width[NDims];
+    child_widths = unique_ptr<point_t>(new point_t());
+    point_t& new_width = *child_widths;
+    for(unsigned int d = 0; d < NDims; d++) {
+        new_width[d] = .5 * boundary.getWidth(d);
+    }
+
     for(unsigned int i = 0; i < no_children; i++) {
         unsigned int div = 1;
+        point_t new_corner;
         for(unsigned int d = 0; d < NDims; d++) {
-            new_width[d] = .5 * boundary.getWidth(d);
-            if((i / div) % 2 == 1) new_corner[d] = boundary.getCorner(d) - .5 * boundary.getWidth(d);
-            else                   new_corner[d] = boundary.getCorner(d) + .5 * boundary.getWidth(d);
+            if((i / div) % 2 == 1)
+                new_corner[d] = boundary.getCorner(d) - new_width[d];
+            else
+                new_corner[d] = boundary.getCorner(d) + new_width[d];
             div *= 2;
         }
-        children[i] = new SPTree(this, data, new_corner, new_width);
+        children[i] = unique_ptr<SPTree>(new SPTree(this, data, move(new_corner), child_widths.get()));
     }
 
     // Move existing points to correct children
     for(unsigned int i = 0; i < size; i++) {
-        bool success = false;
-        for(unsigned int j = 0; j < no_children; j++) {
-            if(!success) success = children[j]->insert(index[i]);
+        for (auto& child : children) {
+            if (child->insert(index[i])) {
+                break;
+            }
         }
-        index[i] = -1;
```

Comment:
This line is unreplicated. Is it unnecessary?

Reply:
Yep.  index is only read for leaf nodes, which this no longer is.